pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -132,6 +132,10 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	pr := original.DeepCopy()
 	pr.Status.InitializeConditions()
 
+	if isDone(&pr.Status) {
+		return nil
+	}
```

Comment:
i think to completely fix #299 we'd want something like this for `TaskRun` too

Reply:
Good point. Will add.